File: dart/luzid-grpc-client/lib/src/client/ping.dart
import 'package:grpc/grpc_connection_interface.dart';
import 'package:luzid_grpc/luzid_grpc.dart';

class PingClient {
  final PingServiceClient _client;

  PingClient(ClientChannelBase channel) : _client = PingServiceClient(channel);

  Future<PingResponse> ping() {
    return _client.ping(PingRequest());
  }
}

File: ts/packages/luzid-grpc-client/src/client/ping.ts
import type { Channel } from '@luzid/grpc-connection'
import { createClient } from '@luzid/grpc-connection'
import {
  PingResponse,
  PingServiceClient,
  PingServiceDefinition,
} from '@luzid/grpc'

export { PingRequest, PingResponse } from '@luzid/grpc'

export class PingClient {
  private readonly client: PingServiceClient

  constructor(channel: Channel) {
    this.client = createClient(PingServiceDefinition, channel)
  }

  async ping(): Promise<PingResponse> {
    return this.client.ping({})
  }
}
